zerver/tests/test_signup.py

```diff
@@ -843,6 +858,52 @@ def test_successful_invite_user_as_guest_from_admin_account(self) -> None:
         self.assertFalse(invitee_profile.is_realm_admin)
         self.assertTrue(invitee_profile.is_guest)
 
+    def test_successful_invite_with_support_param(self) -> None:
+        inviter = self.example_user('iago')
+        self.login(self.example_email('iago'))
+        self.assertTrue(inviter.is_staff)
+
+        invitee = self.nonreg_email('alice')
+        self.invite(invitee, ["Denmark"])
+
+        prereg_user = PreregistrationUser.objects.get(email=invitee)
+        url = self.get_confirmation_url_from_outbox(invitee)
+        url += "?support=True"
+
+        self.assertEqual(prereg_user.status, 0)
+        response = self.client_get(url)
+        prereg_user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(prereg_user.status, 0)
+
+        self.submit_reg_form_for_user(invitee, "password")
+        prereg_user.refresh_from_db()
+        self.assertEqual(prereg_user.status, 0)
+        new_user = UserProfile.objects.get(email=invitee)
+        self.assert_logged_in_user_id(new_user.id)
+
+        # Now test without authentication
```

Comment:
I think it'd be better to have the
```
        self.logout()
        self.assert_logged_in_user_id(None)
```
here, at the very start beginning of this test scenario.

Reply:
@mateuszmandera on placing `self.logout()` at the top, I get the error `Not logged in: API authentication or user session required` at the `self.assert_json_success(self.invite(...))` line.
I haven't noticed any other test case where they invite users when there are no logged in user ids.

Reply:
You're right, it'll be needed to be logged in to do the invite, and log out afterwards. Looks great the way it's done now imo.